modules/browsiRtdProvider.js

```diff
@@ -332,13 +335,23 @@ export const browsiSubmodule = {
   getBidRequestData: setBidRequestsData
 };
 
-function getTargetingData(uc) {
+function getTargetingData(uc, c, us, a) {
   const targetingData = getRTD(uc);
+  const auctionId = a.auctionId
   uc.forEach(auc => {
     if (isNumber(_ic[auc])) {
       _ic[auc] = _ic[auc] + 1;
     }
+    const transactionId = a.adUnits.find(adUnit => adUnit.code === auc).transactionId;
+    events.emit(CONSTANTS.EVENTS.BILLABLE_EVENT, {
```

Comment:
wouldn't this emit for every adUnit code in the auction? is requesting an auction with N ad units supposed to generate N billable events regardless of the bids it gets (or does not get)?

It'd be helpful to have a test case that shows what billable events you expect an auction to generate.

Reply:
Agree, please add actual test cases surrounding these events.

As for when the events fire, perhaps there is or needs to be some documentation on how Browsi determines it? Is it just whenever the viewability is passed into the adServer for a given adUnit?

Reply:
We emit event per adUnitCode, regardless of the bids

Reply:
Added unit testing